src/attributes.jl

```diff
@@ -173,8 +173,13 @@ end
 function write_attribute(attr::Attribute, memtype::Datatype, str::AbstractString)
     strbuf = Base.cconvert(Cstring, str)
     GC.@preserve strbuf begin
-        buf = Base.unsafe_convert(Ptr{UInt8}, strbuf)
-        write_attribute(attr, memtype, buf)
+        if API.h5t_is_variable_str(memtype)
+            ptr = Base.unsafe_convert(Cstring, strbuf)
+            write_attribute(attr, memtype, Ref(ptr))
+        else
+            ptr = Base.unsafe_convert(Ptr{UInt8}, strbuf)
```

Comment:
Do we test this side of the conditional? I suppose this is for a fixed length string?

Reply:
Good point, maybe we should remove this. Don't think we have user-facing API that uses fixed strings